src/components/newDBConnectionPopup/index.js

```diff
@@ -0,0 +1,189 @@
+import React, { useMemo, useCallback } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { CSSTransition } from 'react-transition-group';
+import Input from '../shared/Input';
+import AlertMessage from '../shared/AlertMessage';
+
+
+import { preventDefaultHandler } from '../../utils';
+import styles from '../shared/BasePopup/base.popup.module.scss';
+import '../shared/BasePopup/styles.scss';
+import NewDBCaption from './NewDBCaption';
+import {
+  getVisible,
+  getType,
+  getUsername,
+  getPassword,
+  getHost,
+  getPort,
+  getNameDB,
+  getNameConnection,
+  isError,
+  getErrorMessage,
+} from '../../store/connection/selectors';
+import {
+  newConnectionAction,
+  getHostValue,
+  getPortValue,
+  getPasswordValue,
+  getUsernameValue,
+  getNameDBValue,
+  getTypeValue,
+  getNameConnectionValue,
+  onCloseAction,
+} from '../../store/connection/actions';
+import PopupButtons from '../shared/BasePopup/components/PopupButtons';
+import stylesInput from '../shared/Input/textField.module.scss';
+
+const NewConnection = ({
+  host,
+  port,
+  nameDB,
+  username,
+  password,
+  type,
+  nameConnection,
+  isCreatingInProgress,
+  isVisible,
+  submitForm,
+  isError,
+  errorMessage,
+  okButton,
+  okButtonType,
+  cancelButton,
+  onClose,
+  changeUsername,
+  changePassword,
+  changeHost,
+  changeType,
+  changePort,
+  changeNameConnection,
+  changeNameDB,
+}) => {
+  const formHandler = useMemo(
+    () => preventDefaultHandler(submitForm),
+    [submitForm]
+  );
+  const alertClasses = useMemo(() => [
+    styles.error,
+    isError ? styles.visible : styles.hidden
+  ], [isError]);
+  const backgroundHandler = useCallback((e) => (
+    e.currentTarget === e.target && onClose()
+  ), [onClose]);
+  return (
+    <CSSTransition
+      in={isVisible}
+      timeout={300}
+      classNames="alert"
+      unmountOnExit
+    >
+      <div className={styles.popupWrapper}>
+        <div className={styles.blur} onClick={backgroundHandler} />
+        <div className={`${styles.popup}`}>
+          <div className={styles.header}>
+            <div className={styles.cross} onClick={onClose}>
+              <img src="/images/Popup/cross.png" alt="" />
+            </div>
+          </div>
```

Comment:
You use this code twice in both popups, could you move common code in some base component ?

Reply:
Can't you just use this component for popup layout ? https://github.com/MishaHnatyshyn/Datalyzer/blob/develop/src/components/shared/BasePopup/index.js